### PR TITLE
Keep Index

### PR DESCRIPTION
**Pull request by leogrim**
@ymatsuda 
As for LibraryIndexable, will tweak KeepIndexable's fields depending on the latest product decisions.

### CODE REVIEW

File: kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  val homePageField = "home_page"
+  val recordField = "rec"
+
+  val decoders: Map[String, FieldDecoder] = Map.empty
+}
+
+case class KeepIndexable(keep: Keep) extends Indexable[Keep, Keep] {
+  val id = keep.id.get
+  val sequenceNumber = keep.seq
+  val isDeleted = !keep.isActive
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala:24]: are these variables used, or will be used?

Comment (leogrim) [inline kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala:24]: They are required properties of an Indexable :)

File: kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  override def buildDocument = {
+    import KeepFields._
+    val doc = super.buildDocument
+
+    doc.add(buildKeywordField(libraryField, keep.libraryId.get.toString))
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala:30]: if a keep can exist in multiple libraries, then maybe we can use `buildIteratorField`? not sure about the product design here.

**Comment by ymatsuda** [comment]
Cool! :+1:

File: kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/search/app/com/keepit/search/graph/keep/KeepIndexable.scala:30]: A keep is now uniquely identified as a uri \* library pair, so we can't say that a keep exists in several libraries. A uri may be kept into several libraries and each instance is considered as a different keep.

**Pull request merged**